subsys/net/lib/sockets/Kconfig

```diff
@@ -73,6 +73,31 @@ config NET_SOCKET_MAX_SEND_WAIT
 	  The maximum time a socket is waiting for a blocked connection before
 	  returning an ENOBUFS error.
 
+config NET_SOCKETS_SERVICE
+	bool "Socket service support [EXPERIMENTAL]"
+	select EXPERIMENTAL
+	select EVENTFD
+	help
+	  The socket service can monitor multiple sockets and save memory
+	  by only having one thread listening socket data. If data is received
+	  in the monitored socket, a user supplied work is called.
+	  Note that you need to set CONFIG_NET_SOCKETS_POLL_MAX high enough
+	  so that enough sockets entries can be serviced. This depends on
+	  system needs as multiple services can be activated at the same time
+	  depending on network configuration.
+
+config NET_SOCKETS_SERVICE_STACK_SIZE
+	int "Stack size for the thread handling socket services"
+	default 1024
```

Comment:
Note to self, this needs to be increased to 1200 as the usage was over 90% in my tests.